File: frontend/src/services/cartService.js
import { api } from './api'

export const cartService = {
  getCart: () => api.get('/cart'),

  addItem: (productId, quantity) =>
    api.post('/cart/items', { productId, quantity }),

  updateItem: (productId, quantity) =>
    api.patch(`/cart/items/${productId}`, { quantity }),

  removeItem: (productId) => api.delete(`/cart/items/${productId}`),

  clearCart: () => api.delete('/cart'),
}

File: cart-service/src/main/java/org/n11bootcamp/cartservice/services/CartService.java
package org.n11bootcamp.cartservice.services;



import org.n11bootcamp.cartservice.dtos.requests.AddCartItemRequest;
import org.n11bootcamp.cartservice.dtos.requests.UpdateCartItemRequest;
import org.n11bootcamp.cartservice.models.Cart;

import java.util.UUID;

public interface CartService {

    Cart getCart(UUID userId);

    Cart addItem(UUID userId, AddCartItemRequest request);

    Cart updateItem(UUID userId, UUID productId, UpdateCartItemRequest request);

    void removeItem(UUID userId, UUID productId);

    void clearCart(UUID userId);
}
